frontend/src/components/FragmentComment.js

```diff
@@ -0,0 +1,50 @@
+import '../styles/FragmentComment.css';
+import React from 'react';
+
+function FragmentComment({ children }) {
+  try {
+    const citationRegex = /^\[.*?\]\s*\n(.*)$/m;
```

Comment:
Hmm... Interesting but isn't `?` redundant after `*`?

Reply:
Thanks, it is redundant indeed. This is now fixed in the last commit.